### PR TITLE
fix: Fix `eslint` script/errors and run in PR builds

### PR DESCRIPTION
**Pull request by morgsmccauley**
## Motivation
`eslint` is not enforced anywhere within the code base so having it run within the PR builds helps maintain consistency.

## Description
This PR fixes the `eslint` yarn scripts and adds it to the `pull_request` Github action

## Checklist
- [x] Read the [contributing](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md) guidelines
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [x] Performed a self-review of the PR
- [ ] Added automated tests
- [ ] Manually tested the change

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,8 +60,12 @@
         "doc": "typedoc src && touch docs/.nojekyll",
         "build": "yarn compile",
         "test": "jest test",
-        "lint": "eslint test && eslint src/**/*.ts",
-        "fix": "eslint test --fix && eslint src/**/*.ts --fix",
```
Comment (morgsmccauley) [inline package.json:63]: Prior commands used in `&&` will short-circuit subsequent commands, i.e. if `eslint test` failed, it wouldn't even run `eslint src`.

Using `concurrently` here guarantees that both will be run.

File: src/.eslintrc.yml [hunk 2]
```diff
@@ -8,6 +8,10 @@ extends:
 parser: '@typescript-eslint/parser'
 rules:
   no-inner-declarations: off
+  indent:
+    - error
+    - 4
+    - SwitchCase: 1
```
Comment (morgsmccauley) [inline src/.eslintrc.yml:14]: Without this, `eslint` will enfore:
```js
switch (condition) {
case 'hello':
  return 'world';
default:
  return 'unknown';
}
```
But I feel this is a bit nicer:
```js
switch (condition) {
    case 'hello':
        return 'world';
    default:
        return 'unknown';
}
```

Any oppositions?

**Comment by esaminu** [comment]
Opinions on enforcing these pre-commit / pre-push like on wallet? I assume they're currently not enforced

File: src/.eslintrc.yml (hunk 2 above)
Comment (esaminu) [reply src/.eslintrc.yml:14]: I prefer the latter as well

**Comment by morgsmccauley** [comment]
> Opinions on enforcing these pre-commit / pre-push like on wallet? I assume they're currently not enforced

I find pre-commit a bit invasive at times, so was thinking to add to pre-push but since pushing is usually associated with a PR, and the workflow will now run lint checks I'm wondering if it's even necessary 🤔

**Comment by esaminu** [comment]
My preference is pre commit but seems we did it pre push in the wallet. I think we should do at least one since we want to avoid having to tell external contributors that the CI is failing on their PRs and minimize communication time

**Comment by morgsmccauley** [comment]
@esaminu good point, forgot that actions won't even be run for external contributors initially, so would be good for them to run themselves.

I've added to pre-push 👍🏽

**Review by esaminu — APPROVED** [review]
🚀

**Pull request merged**